web/packages/teleport/src/services/userEvent/types.ts

```diff
@@ -175,13 +175,34 @@ export type DiscoverEventData = DiscoverEventStepStatus & {
   // in the RDS enrollment screen for event
   // tp.ui.discover.database.enroll.rds
   selectedResourcesCount?: number;
+
+  // serviceDeploy is only considered for 'tp.ui.discover.deployService'
+  // event and describes how an agent got deployed.
+  serviceDeploy?: DiscoverServiceDeploy;
 };
 
 export type DiscoverEventStepStatus = {
   stepStatus: DiscoverEventStatus;
   stepStatusError?: string;
 };
 
+export type DiscoverServiceDeploy = {
+  method: DiscoverServiceDeployMethod;
+  type: DiscoverServiceDeployType;
+};
+
+export enum DiscoverServiceDeployMethod {
+  Unspecified = 'DEPLOY_METHOD_UNSPECIFIED',
+  Auto = 'DEPLOY_METHOD_AUTO',
+  Manual = 'DEPLOY_METHOD_MANUAL',
+}
+
+export enum DiscoverServiceDeployType {
+  Unspecified = 'DEPLOY_TYPE_UNSPECIFIED',
+  InstallScript = 'DEPLOY_TYPE_INSTALL_SCRIPT',
+  AmazonEcs = 'DEPLOY_TYPE_AMAZON_ECS',
```

Comment:
This should be `AmazonECS` unless we need it to be `Ecs` for some reason

Reply:
we don't all cap acronyms atm in our frontend